packages/interaction-kit/src/interactions/index.ts

```diff
@@ -219,16 +214,21 @@ export function handler(
 		}
 
 		case InteractionType.MessageComponent: {
-			const component = application.getComponent(json.data.custom_id);
+			let component = application.getComponent(json.data.custom_id);
+
+			if (component == null) {
+				component = await application.findComponent(json.data.custom_id);
+			}
```

Comment:
```suggestion
			const component = application.getComponent(json.data.custom_id) 
				?? await application.findComponent(json.data.custom_id);
```
I wasn't able to push directly :P